models/src/node.rs

```diff
@@ -0,0 +1,213 @@
+use crate::constants;
+
+use async_trait::async_trait;
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::OwnerReference;
+use kube::api::{Api, ObjectMeta, Patch, PatchParams, PostParams};
+use kube::CustomResource;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use snafu::{ResultExt, Snafu};
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub")]
+pub enum BottlerocketNodeError {
+    #[snafu(display(
+        "Unable to create BottlerocketNode ({}, {}): {}",
+        selector.node_name,
+        selector.node_uid,
+        source
+    ))]
+    CreateBottlerocketNode {
+        source: Box<dyn std::error::Error>,
+        selector: BottlerocketNodeSelector,
+    },
+
+    #[snafu(display(
+        "Unable to update BottlerocketNode status ({}, {}): {}",
```

Comment:
Nit: I remember folks usually ask to put error messages between `''`:

```suggestion
        "Unable to update BottlerocketNode status ({}, {}): '{}'",
```